lib/transactional.pm

```diff
@@ -111,8 +111,9 @@ sub process_reboot {
                 unlock_if_encrypted();
             }
             # Replace by wait_boot if possible
-            assert_screen 'grub2', 150;
-            wait_screen_change { send_key 'ret' };
+            if (check_screen 'grub2', 150) {
+                wait_screen_change { send_key 'ret' };
```

Comment:
maybe check the screen for grub and Linux-login? simply skip the send key if we're already at the login screen.

Reply:
In fact, some my tests failed in reboot_on_changes, and alp need send_key ret still